Monika After Story/game/script-story-events.rpy

```diff
@@ -269,6 +269,202 @@ label monika_changename:
+                    # jumping here to pass to set the vars that all 3 methods of setting your bday need
+                    jump mas_bday_player_bday_select_select
+            "No":
+                m 3rksdlc "Oh! Okay then..."
+                m 1eksdld "When {i}is{/i} your birthdate, [player]?"
+                jump mas_bday_player_bday_select_select
+
+    else:
+        m 3wud "I don't actually know when your birthdate is!"
+        m 3hub "That's something I should probably know, ahaha!"
+        m 1eua "So, when were you born, [player]?"
+        jump mas_bday_player_bday_select_select
+
+label birthdate_set:
+    python:
+        bday_upset_ev = mas_getEV('mas_player_bday_upset_minus')
+        if bday_upset_ev is not None:
+            bday_upset_ev.start_date = mas_player_bday_curr
+            bday_upset_ev.end_date = mas_player_bday_curr + datetime.timedelta(days=1)
+            bday_upset_ev.conditional = (
+                "mas_isplayer_bday() "
+                "and not persistent._mas_player_bday_decor "
+                "and not persistent._mas_player_bday_in_player_bday_mode "
+                "and not persistent._mas_player_bday_low_aff "
+                "and not persistent._mas_player_bday_when_found")
+            bday_upset_ev.action = EV_ACT_QUEUE
+            Event._verifyAndSetDatesEV(bday_upset_ev)
+            
+        bday_no_restart_ev = mas_getEV('mas_player_bday_no_restart')
+        if bday_no_restart_ev is not None:
+            bday_no_restart_ev.start_date = datetime.datetime.combine(mas_player_bday_curr, datetime.time(hour=19))
+            bday_no_restart_ev.end_date = mas_player_bday_curr + datetime.timedelta(days=1)
+            bday_no_restart_ev.conditional = (
+                "mas_isplayer_bday() "
+                "and not persistent._mas_player_bday_in_player_bday_mode "
+                "and not persistent._mas_player_bday_decor "
+                "and not persistent._mas_player_bday_when_found "
+                "and not persistent._mas_player_bday_low_aff "
+                "and not mas_isO31() "
+                "and not mas_isD25()")
+            bday_no_restart_ev.action = EV_ACT_QUEUE
+            Event._verifyAndSetDatesEV(bday_no_restart_ev)
+    
+        bday_holiday_ev = mas_getEV('mas_player_bday_other_holiday')
+        if bday_holiday_ev is not None:
+            bday_holiday_ev.start_date = mas_player_bday_curr
+            bday_holiday_ev.end_date = mas_player_bday_curr + datetime.timedelta(days=1)
+            bday_holiday_ev.conditional = (
+                "mas_isplayer_bday() "
+                "and not persistent._mas_player_bday_in_player_bday_mode "
+                "and not persistent._mas_player_bday_decor "
+                "and not persistent._mas_player_bday_when_found "
+                "and not persistent._mas_player_bday_low_aff "
+                "and "
+                "mas_isO31() or mas_isD25()")
```

Comment:
your last condition here shoulld be `and (mas_isO31() or mas_isD25())` since its clear that those should be grouped.